File: app/src/main/java/com/cleanup/todoc/datasource/Entity/ProjectEntity.java
package com.cleanup.todoc.datasource.Entity;


import androidx.annotation.ColorInt;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.List;

/**
 * <p>Entity for project in which tasks are included.</p>
 *
 * @author Xavier Carpentier
 */
@Entity(tableName = "ProjectEntity")
public class ProjectEntity {
    /**
     * The unique identifier of the project
     */
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    private final long id;

    /**
     * The name of the project
     */
    @NonNull
    @ColumnInfo(name = "name")
    private final String name;

    /**
     * The hex (ARGB) code of the color associated to the project
     */
    @ColumnInfo(name = "color")
    @ColorInt
    private final int color;

    /**
     * Instantiates a new Project.
     *
     * @param id    the unique identifier of the project to set
     * @param name  the name of the project to set
     * @param color the hex (ARGB) code of the color associated to the project to set
     */
    public ProjectEntity(long id, @NonNull String name, @ColorInt int color) {
        this.id = id;
        this.name = name;
        this.color = color;
    }


    /**
     * Returns the unique identifier of the project.
     *
     * @return the unique identifier of the project
     */
    public long getId() {
        return id;
    }

    /**
     * Returns the name of the project.
     *
     * @return the name of the project
     */
    @NonNull
    public String getName() {
        return name;
    }

    /**
     * Returns the hex (ARGB) code of the color associated to the project.
     *
     * @return the hex (ARGB) code of the color associated to the project
     */
    @ColorInt
    public int getColor() {
        return color;
    }

}

File: app/src/main/java/com/cleanup/todoc/datasource/entity/ProjectEntity.kt
package com.cleanup.todoc.datasource.entity

import androidx.annotation.ColorInt
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 *
 * Entity for project in which tasks are included.
 *
 * @author Xavier Carpentier
 */
@Entity(tableName = "ProjectEntity")
data class ProjectEntity(
    /**
     * The unique identifier of the project
     */
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long,

    /**
     * The name of the project
     */
    @ColumnInfo(name = "name")
    val name: String,

    /**
     * The hex (ARGB) code of the color associated to the project
     */
    @ColumnInfo(name = "color")
    @ColorInt
    val color: Int
)
